### PR TITLE
Return Appropriate Error Message when a Web Action is Invoked with an Unsupported Content-type

### PR DESCRIPTION
**Pull request by dubeejw**
- Inform user's when an unsupported content-type is passed to a Web Action

Closes: https://github.com/openwhisk/openwhisk/issues/1886

### CODE REVIEW

File: common/scala/src/main/scala/whisk/http/ErrorResponse.scala [hunk 1]
```diff
@@ -114,8 +114,8 @@ object Messages {
     /** Error for meta api. */
     val propertyNotFound = "Response does not include requested property."
     def invalidMedia(m: MediaType) = s"Response is not valid '${m.value}'."
-    val contentTypeNotSupported = """Content type must be specified and one of ["json", "html", "http", "text"]."""
-
+    val contentTypeExtentionNotSupported = """Content type must be specified and one of ["json", "html", "http", "text"]."""
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/http/ErrorResponse.scala]: while you're at it - mind changing ` contentTypeExtentionNotSupported = """Content type must` to ` contentTypeExtentionNotSupported = """Extension must`?

maybe also prefix each of the extensions with a `.` (dot).

File: common/scala/src/main/scala/whisk/http/ErrorResponse.scala [hunk 2]
```diff
@@ -114,8 +114,8 @@ object Messages {
...
+    val contentTypeNotSupported = "Content type must be appication/json or application/x-www-form-urlencoded."
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/http/ErrorResponse.scala]: typo - maybe better to use "`application/json`" from `MediaTypes` instead.

File: core/controller/src/main/scala/whisk/core/controller/Meta.scala [hunk 3]
```diff
@@ -330,7 +330,7 @@ trait WhiskMetaApi
                     body => process(body)
                 } ~ entity(as[FormData]) {
                     form => process(Some(form.fields.toMap.toJson.asJsObject))
-                }
+                } ~ terminate(BadRequest, Messages.contentTypeNotSupported)
```
Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/controller/Meta.scala]: please add a test.

**Comment by dubeejw** [comment]
PG2 1120, approved.

**Comment by rabbah** [comment]
Nice. Checking the issue again I think we covered the ingress but not an error on egress: https://github.com/openwhisk/openwhisk/issues/1886#issuecomment-282353511 we should add one more test where accept header is set to an incompatible type (json and return text).

**Comment by rabbah** [comment]
superseded by #2004.

**Pull request closed without merging**